staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go

```diff
@@ -1005,3 +1005,88 @@ const (
 	LabelSelectorOpExists       LabelSelectorOperator = "Exists"
 	LabelSelectorOpDoesNotExist LabelSelectorOperator = "DoesNotExist"
 )
+
+type VersionedFieldSet struct {
+	// APIVersion defines the versioned schema of this representation of an object.
+	// Servers should convert recognized schemas to the latest internal value, and
+	// may reject unrecognized values.
+	// More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#resources
+	APIVersion string `json:"apiVersion,omitempty" protobuf:"bytes,1,opt,name=apiVersion"`
+	// Fields TODO
+	Fields FieldSet `json:"fields,omitempty" protobuf:"bytes,2,opt,name=fields,casttype=FieldSet"`
+}
+
+// FieldSet identifies a set of fields.
+type FieldSet struct {
+	// Members lists fields that are part of the set.
+	Members []FieldPathElement `json:"members" protobuf:"bytes,1,rep,name=members"`
+	// Children lists child fields which themselves have children that are
+	// members of the set. Appearance in this list does not imply membership.
```

Comment:
Would `, although it does imply some ancestor field is a member` make it clearer?

Reply:
ancestor?

Reply:
Yes sorry, descendent